src/OrchardCore/OrchardCore.DisplayManagement/Theming/ThemeLayout.cs

```diff
@@ -16,15 +20,36 @@ public class ThemeLayout : Razor.RazorPage<dynamic>
     {
         public override async Task ExecuteAsync()
         {
-            // The View's body is rendered 
+            // The View's body is rendered
             var body = RenderLayoutBody();
 
             if (ThemeLayout != null)
             {
                 // Then is added to the Content zone of the Layout shape
                 ThemeLayout.Content.Add(body);
 
-                // Finally we render the Shape's HTML to the page's output
+                // Render Shapes in Content
+                if(ThemeLayout.Content is Shape content)
+                {
+                    var htmlContent = await DisplayAsync(content);
+                    ThemeLayout.Content = htmlContent;
+                }
+
+                if (ThemeLayout is ZoneHolding layout)
+                {
+                    foreach (var zone in layout.Properties.ToArray())
+                    {
+                        if((zone.Value is Shape shape) &&
```

Comment:
Idem here

Then below we could do a case insensitive check on `Content`, as we use it above as a dynamic property.

But in fact we don't need to check if the zone name is `Content` because we check if it is an `IShape`, and above if it was an `IShape` here it's no more the case, it is an `IHtmlContent`.

So finally we can just use `if (zone.Value is IShape shape)`

Reply:
changed to IShape